### DISCUSSION TITLE
Flutter Modular em SPA, é possível?

### DISCUSSION THREAD

**Discussion by alexsandercdm** [Flutterando/forum#97] [Perguntas e Respostas] [answered]
Boa tarde, pessoal!

Espero que estejam todos bem e codando, estou trabalhando em um projeto Web, cuja estratégia de Interações entre páginas será, basicamente por SPA. Em alguns testes que realizei verifiquei que não funcionou da forma que estava esperando.

O **Modular.to.navigate('/route')** sempre, direciona para uma nova pagina, atualizando todos os meus elementos e assim derrubando minhas expectativas em utilizar o modular neste projeto.

Alguém poderia me ajudar nesta questão? Ou o modular realmente não atende a essa estratégia? Verifiquei o go_router que resolve meu “problema”, mas outras não.

**Comment by eduardoflorence** [comment, marked as answer]
Olá @alexsandercdm,
    Você tem que utilizar uma feature do Modular chamada RouterOutlet.
    Aqui tem a documentação e um exemplo parecido com o que você quer: https://modular.flutterando.com.br/docs/flutter_modular/navegation#routeroutlet

**Reply by alexsandercdm** [reply]
Muito obrigado, foi de grande valia.
